test/e2e/clusterclaim_test.go

```diff
@@ -1,108 +1,126 @@
+		originalValue := claim.Spec.Value
 
 		// modify the the claim value
-		err = unstructured.SetNestedField(claim.Object, rand.String(6), "Spec", "Value")
-		gomega.Expect(err).ToNot(gomega.HaveOccurred())
-		_, err = util.UpdateClusterResource(dynamicClient, clusterClaimGVR, claim)
+		err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+			claim, err = clusterClient.ClusterV1alpha1().ClusterClaims().Get(context.TODO(), claimName, v1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			claim.Spec.Value = rand.String(6)
+			_, err = clusterClient.ClusterV1alpha1().ClusterClaims().Update(context.TODO(), claim, v1.UpdateOptions{})
+			return err
+		})
 		gomega.Expect(err).ToNot(gomega.HaveOccurred())
 
 		// check if the change is rollback
-		gomega.Eventually(func() bool {
-			claim, err = util.GetClusterResource(dynamicClient, clusterClaimGVR, claimName)
+		gomega.Eventually(func() error {
+			claim, err = clusterClient.ClusterV1alpha1().ClusterClaims().Get(context.TODO(), claimName, v1.GetOptions{})
 			if err != nil {
-				return false
+				return err
+			}
+			if originalValue != claim.Spec.Value {
+				return fmt.Errorf("the claim is not rollback")
 			}
+			return nil
+		}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
+	})
 
-			value, _, err := unstructured.NestedString(claim.Object, "Spec", "Value")
+	ginkgo.It("should sync the label to claim", func() {
+		var err error
+		// add label to managedCluster
+		err = util.UpdateManagedClusterLabels(clusterClient, managedClusterName, map[string]string{"test": "test"})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+		// should get the claim synced from label
+		gomega.Eventually(func() error {
+			_, err := clusterClient.ClusterV1alpha1().ClusterClaims().Get(context.TODO(), "test", v1.GetOptions{})
+			return err
+		}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
+
+		// delete label from managedCluster
+		err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+			cluster, err := clusterClient.ClusterV1().ManagedClusters().Get(context.TODO(), managedClusterName, v1.GetOptions{})
 			if err != nil {
-				return false
+				return err
 			}
+			delete(cluster.Labels, "test")
+			_, err = clusterClient.ClusterV1().ManagedClusters().Update(context.TODO(), cluster, v1.UpdateOptions{})
+			return err
+		})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
 
-			return originalValue == value
-		}, eventuallyTimeout, eventuallyInterval).Should(gomega.BeTrue())
+		// should delete the claim synced from label
+		gomega.Eventually(func() error {
+			_, err := clusterClient.ClusterV1alpha1().ClusterClaims().Get(context.TODO(), "test", v1.GetOptions{})
+			if errors.IsNotFound(err) {
+				return nil
+			}
+			return fmt.Errorf("failed delete claim syned from label %v", err)
+		}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
```

Comment:
should we add an e2e to ensure user created claim will not be removed?

Reply:
done

Reply:
create a clusterClaim, test if get it in 1 min